File: 287-find-the-duplicate-number/find-the-duplicate-number.java
class Solution {
    static{
        for(int i=0 ;i<500;i++){
            findDuplicate(new int[]{0});
        }
    }
    public static int findDuplicate(int[] nums) {
        int[] dp = new int[nums.length];
        for(int i:nums) {
            if(dp[i] == -1) return i;
            dp[i] = -1;
        }
        return -1;
    }
}

File: 287-find-the-duplicate-number/find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        dp = [0]*len(nums)
        for i in range(0, len(nums)):
            if dp[nums[i]] == -1:
                return nums[i]

            dp[nums[i]] = -1
            
